src/frontend/metrics/generateMetricsReport.test.ts

```diff
@@ -1,38 +1,46 @@
 import * as path from 'node:path';
 import * as fs from 'node:fs';
-import generateMetricsReport from './generateMetricsReport';
+import {
+  generateMetricsReport,
+  getPercentageOfNetworkAvailability,
+} from './generateMetricsReport';
+import {generate} from '@mapeo/mock-data';
+import positionToCountries from './positionToCountries';
+import type {Observation} from '@mapeo/schema';
+import {addToSet} from './utils';
 
 type MetricsReportOptions = Parameters<typeof generateMetricsReport>[0];
 
 describe('generateMetricsReport', () => {
   const packageJson = readPackageJson();
+  const count = 10;
+  const n = Math.floor(Math.random() * count);
+  const observations: ReadonlyArray<Observation | Record<string, never>> = [
+    ...generate('observation', {count}).map((obs, idx) =>
+      // Manually add Machias Seal Island, disputed territory
+      idx === n ? {...obs, lat: 44.5, lon: -67.101111} : obs,
+    ),
+    {},
+  ];
+  const generatedCountries = new Set<string>();
+
+  for (const {lat, lon} of observations) {
+    if (typeof lat === 'number' && typeof lon === 'number') {
+      addToSet(generatedCountries, positionToCountries(lat, lon));
+    }
+  }
+
+  const generatedNetworkAvailabilityPercentage =
+    getPercentageOfNetworkAvailability(observations);
```

Comment:
I'm a little uneasy about using this in tests. If `getPercentageOfNetworkAvailability` is wrong, the test will still pass.

Would you mind adding tests for `getPercentageOfNetworkAvailability` specifically, that don't use mock data? I think that would make me more comfortable with this part of the test.

Reply:
Do you think this is also true with `generatedCountries`? I'm basically generating mock-data and matching tests with that

Reply:
I've created individual tests for networkAvailability without mock data. I think maybe thats enough?